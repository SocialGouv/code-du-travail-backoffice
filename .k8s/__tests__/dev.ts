import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  process.env.HARBOR_PROJECT = "cdtn";
  process.env.KUBE_NAMESPACE = "cdtn-backoffice";
  expect(
    await getEnvManifests("dev", "", {
      ...project("code-du-travail-backoffice").dev,
      RANCHER_PROJECT_ID: "c-bar:p-foo",
    }),
  ).toMatchSnapshot();
});
