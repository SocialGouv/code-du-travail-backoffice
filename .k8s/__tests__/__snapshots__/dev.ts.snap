// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    janitor/ttl: 7d
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-f8qps:p-46tj7
    git/branch: refs/heads/mybranch
    git/remote: socialgouv/code-du-travail-backoffice
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: code-du-travail-backoffice
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: code-du-travail-backoffice-mybranch
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: app
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: app
  namespace: code-du-travail-backoffice-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/code-du-travail-backoffice
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: app
        application: code-du-travail-backoffice
        component: code-du-travail-backoffice
        owner: code-du-travail-backoffice
        team: code-du-travail-backoffice
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/cdtn/cdtn-backoffice/app:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 20
            timeoutSeconds: 5
          name: app
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 20
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 512Mi
            requests:
              cpu: 100m
              memory: 256Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - configMapRef:
                name: app-configmap
      initContainers:
        - name: wait-for-postgrest
          image: bash:4
          imagePullPolicy: Always
          resources:
            requests:
              cpu: 5m
              memory: 16Mi
            limits:
              cpu: 20m
              memory: 32Mi
          command:
            - sh
            - '-c'
            - >-
              until nslookup postgrest.$(cat
              /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local;
              do echo en attente de postgrest; sleep 2; done
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: app-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  namespace: code-du-travail-backoffice-mybranch
data:
  CDTN_API_URL: https://cdtn-api.fabrique.social.gouv.fr
  NODE_ENV: production
  POSTGREST_URI: http://postgrest
  APP_PORT: '3000'
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: app
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: app
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: code-du-travail-backoffice-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: app
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: app
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: app
  namespace: code-du-travail-backoffice-mybranch
spec:
  rules:
    - host: code-du-travail-backoffice-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: app
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - code-du-travail-backoffice-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: batch/v1
kind: Job
spec:
  backoffLimit: 5
  template:
    spec:
      containers:
        - command:
            - create-db-user
          env:
            - name: NEW_DB_NAME
              value: autodevops_mybranch
            - name: NEW_USER
              value: user_mybranch
            - name: NEW_PASSWORD
              value: password_mybranch
            - name: NEW_DB_EXTENSIONS
              value: hstore pgcrypto citext uuid-ossp
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: ghcr.io/socialgouv/docker/azure-db:6.56.1
          imagePullPolicy: IfNotPresent
          name: create-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400
metadata:
  annotations:
    kapp.k14s.io/update-strategy: skip
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: create-db-job-mybranch
  namespace: code-du-travail-backoffice-mybranch
---
apiVersion: v1
kind: Secret
stringData:
  DATABASE_URL: >-
    postgresql://user_mybranch%40codedutravailbackofficedevserver.postgres.database.azure.com:password_mybranch@codedutravailbackofficedevserver.postgres.database.azure.com/autodevops_mybranch?sslmode=require
  DB_URI: >-
    postgresql://user_mybranch%40codedutravailbackofficedevserver.postgres.database.azure.com:password_mybranch@codedutravailbackofficedevserver.postgres.database.azure.com/autodevops_mybranch?sslmode=require
  HASURA_GRAPHQL_DATABASE_URL: >-
    postgresql://user_mybranch%40codedutravailbackofficedevserver.postgres.database.azure.com:password_mybranch@codedutravailbackofficedevserver.postgres.database.azure.com/autodevops_mybranch?sslmode=require
  PGDATABASE: autodevops_mybranch
  PGHOST: codedutravailbackofficedevserver.postgres.database.azure.com
  PGPASSWORD: password_mybranch
  PGRST_DB_URI: >-
    postgresql://user_mybranch%40codedutravailbackofficedevserver.postgres.database.azure.com:password_mybranch@codedutravailbackofficedevserver.postgres.database.azure.com/autodevops_mybranch?sslmode=require
  PGSSLMODE: require
  PGUSER: user_mybranch@codedutravailbackofficedevserver.postgres.database.azure.com
metadata:
  annotations:
    kapp.k14s.io/update-strategy: skip
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/code-du-travail-backoffice
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: code-du-travail-backoffice
    component: code-du-travail-backoffice
    owner: code-du-travail-backoffice
    team: code-du-travail-backoffice
    cert: wildcard
  name: azure-pg-user-mybranch
  namespace: code-du-travail-backoffice-mybranch
---
"
`;
